lib/utils/net-utils.js

```diff
@@ -4,29 +4,26 @@ const Promise = require('bluebird')
 const net = require('net')
 
 const NetUtils = {
-  isPortAvailable (port, listenAddress) {
-    return new Promise((resolve, reject) => {
-      const server = net.createServer()
+  getAvailablePort (port, listenAddress) {
+    function listen (port, listenAddress) {
+      return new Promise((resolve, reject) => {
+        const server = net.createServer()
 
-      server.unref()
-      server.on('error', (err) => {
-        server.close()
-        if (err.code === 'EADDRINUSE' || err.code === 'EACCES') {
-          resolve(false)
-        } else {
-          reject(err)
-        }
-      })
+        server.on('error', (err) => {
+          server.close()
+          if (err.code === 'EADDRINUSE' || err.code === 'EACCES') {
+            server.listen(++port, listenAddress)
+          } else {
+            reject(err)
+          }
+        })
 
-      server.listen(port, listenAddress, () => {
-        server.close(() => resolve(true))
+        server.listen(port, listenAddress, () => {
+          resolve([port, server])
```

Comment:
I believe you can get the port as `server.address().port`: https://nodejs.org/api/net.html#net_server_address
So maybe no need to pass it as a param.

Reply:
@sergei-startsev What do you think about that ?